src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -175,6 +175,28 @@ uct_ib_mlx5_inline_copy(void *restrict dest, const void *restrict src, unsigned
 }
 
 
+/**
+ * Copy uct_iov_t array to inline segment, taking into account QP wrap-around.
+ *
+ * @param dest     Inline data in the WQE to copy to.
+ * @param iov      A pointer to an array of uct_iov_t elements.
+ * @param iov_cnt  A number of elements in a iov array.
+ * @param wq       Send work-queue.
+ */
+static UCS_F_ALWAYS_INLINE void
+uct_ib_mlx5_inline_iov_copy(void *restrict dest, const uct_iov_t *iov,
+                            size_t iovcnt, uct_ib_mlx5_txwq_t *wq)
+{
+    size_t copied;
+
+    ucs_assert(dest != NULL);
+
+    copied = uct_iov_to_buffer(iov, iovcnt, 0, dest,
+                               UCS_PTR_BYTE_DIFF(dest, wq->qend));
+    uct_iov_to_buffer(iov, iovcnt, copied, wq->qstart, SIZE_MAX);
```

Comment:
in most cases, there is no wraparound, so IMO better like this:
if (ucs_likely(iov_length <= UCS_PTR_BYTE_DIFF...) {
   uct_iov_to_buffer(iov, iovcnt, 0, dest, SIZE_MAX); /* fast path, compiler can remove some checks */
} else {
  /* 2-part copy */
}